Zend/zend_compile.c

```diff
@@ -3704,14 +3704,18 @@ static void zend_compile_assert(znode *result, zend_ast_list *args, zend_string
 		}
 		opline->result.num = zend_alloc_cache_slot();
 
-		if (args->children == 1 &&
-		    (args->child[0]->kind != ZEND_AST_ZVAL ||
-		     Z_TYPE_P(zend_ast_get_zval(args->child[0])) != IS_STRING)) {
-			/* add "assert(condition) as assertion message */
+		if (args->children == 1) {
+			/* add "assert(condition)" as default assertion description */
```

Comment:
This chunk looks right to me.
The original condition looks strange after "assert() with string argument" was removed.
@nikic May be "assert() with string argument" should cause compilation error now?

Now assert("0") throws, but assert("false") does not.

Reply:
Great :) I also felt that it looked strange.